### PR TITLE
Stop compiling fragile NGen support into the runtime

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by alpencolt** [comment]
@jkotas are you going to remove fragile mode in 3.0 release?
Or it still but under `#ifdef`?

**Comment by jkotas** [comment]
We will keep the code under ifdef.

**Comment by mjsabby** [comment]
@jkotas all the relative ptr stuff would still remain after this PR finishes? I don't see any changes to those things.

**Comment by jkotas** [comment]
>  all the relative ptr stuff would still remain after this PR finishes? 

It won't be in the compiled binary. The ifdef that this PR is flipping turns the relative pointers into regular pointers.

**Comment by jkotas** [comment]
This change makes microbenchmarks exercising paths where fragile NGen support was on a hot path faster. For example, casting `List<int>` to `IList<string>` using `as` is 20% faster on Windows x64.

This will also delete peanut butter cost spread over the runtime. CoreCLR.dll is ~200kB smaller on Windows x64.

File: clrdefinitions.cmake [hunk 1]
```diff
@@ -175,7 +175,6 @@ endif(CLR_CMAKE_PLATFORM_UNIX)
 if(CLR_CMAKE_PLATFORM_LINUX OR CLR_CMAKE_PLATFORM_FREEBSD)
     add_definitions(-DFEATURE_PERFMAP)
 endif(CLR_CMAKE_PLATFORM_LINUX OR CLR_CMAKE_PLATFORM_FREEBSD)
-add_definitions(-DFEATURE_PREJIT)
```
Comment (jkotas) [inline clrdefinitions.cmake]: `FEATURE_PREJIT` means fragile NGen support in the runtime after this change.

File: crossgen.cmake [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Contains the crossgen build specific definitions. Included by the leaf crossgen cmake files.
 
 add_definitions(
+    -DFEATURE_PREJIT
```
Comment (jkotas) [inline crossgen.cmake]: I have not bothered with fixing crossgen to compile with `FEATURE_PREJIT` undefined.

**Comment by sergiy-k** [comment]
Thank you @jkotas! LGTM.

**Pull request merged**